### DISCUSSION TITLE
RAG Document Chat Not Working with Agents

### DISCUSSION THREAD

**Discussion by ogipogi** [danny-avila/LibreChat#4992] [Troubleshooting] [answered]
### What happened?

When using agents with RAG (Retrieval-Augmented Generation), the AI chat doesn't receive the file content when asking questions, even though document upload works successfully. This is due to a conditional check in the code that prevents context handlers from being created when using agents.

### Steps to Reproduce:
1. Configure and enable RAG
2. Configure and enable agents
3. Start a chat with an agent
4. Upload a document to the chat
5. Ask questions about the document content
6. Observe that the agent responses don't include information from the uploaded document

### Expected behavior:
The agent should be able to access and utilize the content from uploaded documents when responding to questions, similar to how it works with non-agent endpoints.

### Actual behavior:
The agent cannot access the document content due to the `isAgentsEndpoint` condition blocking the creation of context handlers.

### Code location:
https://github.com/danny-avila/LibreChat/blob/f15035542f7668254691f2692539a12b799f6585/api/server/controllers/agents/client.js#L280-L285

### Workaround:
Remove the `!isAgentsEndpoint(this.options.endpoint)` condition to allow context handlers to be created for agent endpoints:
```javascript
if (this.message_file_map) {
  this.contextHandlers = createContextHandlers(
    this.options.req,
    orderedMessages[orderedMessages.length - 1].text,
  );
}
```

- LibreChat version: main #c2e0789a

Thanks for checking 🙏 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
For agents, it is intended to utilize the RAG API with the File Search capability: https://www.librechat.ai/docs/features/agents#file-search

**Reply by ogipogi** [reply]
That means, is it also for documents that the user is uploading?

<img width="446" alt="Screenshot 2024-12-14 at 19 27 26" src="https://github.com/user-attachments/assets/37834ae3-3749-490d-9c2f-72cd465c8eea" />

I thought that file search is for static files uploaded upfront that agent can access...

**Reply by danny-avila** [reply, marked as answer]
Yes, that will allow the user to upload files, for which the agent will be aware of them if uploaded correctly: 
![firefox_CjinaQ7qey](https://github.com/user-attachments/assets/edd342f7-52ac-47b5-b77d-5e6ac665119c)

**Reply by ogipogi** [reply]
Thanks for the insights, I tested it and it worked now 🙏
